tests/conftest.py

```diff
@@ -1,8 +1,1322 @@
 import warnings
 import pytest
 
+import numpy as np
 
-@pytest.fixture(scope='session', autouse=True)
+from tests.get_input import get_known_input
+
+
+@pytest.fixture(scope="session", autouse=True)
 def session_setup():
-    warnings.filterwarnings('once', category=DeprecationWarning)
-    warnings.filterwarnings('once', category=PendingDeprecationWarning)
+    warnings.filterwarnings("once", category=DeprecationWarning)
+    warnings.filterwarnings("once", category=PendingDeprecationWarning)
+
+
+def _skip_if_no_petsc_slepc() -> bool:
+    try:
+        import petsc4py
+        import slepc4py
+        return False
+    except ImportError:
+        return True
+
+    
+skip_if_no_petsc_slepc = pytest.mark.skipif(
+    _skip_if_no_petsc_slepc(), reason="No PETSc or SLEPc is installed."
+)
+
+
+@pytest.fixture(scope="session")
+def count_sd():
+    return np.array(
+        [
+            2.1196820476928461e-03,
+            5.4491826226066090e-03,
+            2.3426486027095936e-02,
+            5.8221266809978502e-02,
+            7.2479128130780376e-02,
+            5.4041893715942611e-02,
+            2.8355746638004300e-02,
+            1.4687796830475429e-02,
+            9.4685797130430443e-03,
+            5.8791181322801582e-03,
+            5.1292306154076886e-03,
+            4.1493775933609959e-03,
+            3.6094585812128182e-03,
+            3.7094435834624808e-03,
+            4.4093385992101186e-03,
+            5.5791631255311704e-03,
+            7.3189021646752990e-03,
+            1.3967904814277858e-02,
+            3.0445433185022245e-02,
+            5.9751037344398343e-02,
+            8.7266909963505473e-02,
+            8.8306753986901965e-02,
+            6.0670899365095239e-02,
+            2.9825526171074340e-02,
+            1.3857921311803230e-02,
+            7.7488376743488473e-03,
+            5.3791931210318451e-03,
+            3.9594060890866369e-03,
+            4.0093985902114682e-03,
+            3.2395140728890665e-03,
+            3.7694345848122784e-03,
+            5.3691946208068789e-03,
+            5.8091286307053944e-03,
+            7.6788481727740835e-03,
+            1.4617807328900665e-02,
+            2.7865820126980953e-02,
+            5.3891916212568114e-02,
+            7.4298855171724243e-02,
+            5.8661200819877017e-02,
+            2.2656601509773534e-02,
+            5.9291106334049895e-03,
+            2.9895515672649104e-03,
+        ],
+        dtype=np.float64,
+    )
+
+
+@pytest.fixture(scope="session")
+def count_A():
+    return np.array(
+        [
+            [2.8405568184913244e-01, 2.8755576853720538e-01, 4.2838854961366207e-01],
+            [3.7905560917700676e-01, -3.3892409425382630e-01, -4.0131514923180477e-02],
+            [-1.9965336957541363e-01, -2.7103148561320506e-01, 4.7068485518861869e-01],
+        ],
+        dtype=np.float64,
+    )
+
+
+@pytest.fixture(scope="session")
+def count_A_sparse():
```

Comment:
I'd guess different impl. for sparse/dense in the Schur decomposition. But apart from signs, they are close-ish.

Reply:
For me those differences would be troublesome. Are you sure you are happy with those (since you are using sparse matrices a lot and this would unnerve me, if I were you.)

Reply:
Think @Marius1311 knows better.